File: src/python/ch01/1.2/l2/demo_multiplication.py
def naive_mul(x: int, y: int) -> int:
    """Use Python's * as baseline (Python already uses optimized methods)."""
    return x * y

def main():
    a, b = 1234, 5678
    print("Naive multiplication:", naive_mul(a, b))
    print("Builtin multiplication:", a * b)

if __name__ == "__main__":
    main()

File: src/erlang/ch01/1.2/src/demo_multiplication.erl
-module(demo_multiplication).
-export([main/0]).

naive_mul(X, Y) ->
    X * Y.

main() ->
    A = 1234, B = 5678,
    io:format("Naive multiplication: ~p~n", [naive_mul(A,B)]),
    io:format("Builtin multiplication: ~p~n", [A*B]).
